### PR TITLE
Added Logic to Download Model from HugginFace

### PR DESCRIPTION
**Pull request by lu-ohai**
### Added Logic to Download Model from HugginFace

- Added `download_from_hf: Optional[bool] = True` and `local_dir: Optional[str] = None` to `ImportModelDetails`
- Added logic to download and validate model from hugginface

### Notebook

- SAFETENSORS

Screenshot 2024-07-29 at 3 29 52 PM
Screenshot 2024-07-29 at 3 30 12 PM

- GGUF

Screenshot 2024-07-29 at 3 37 24 PM
Screenshot 2024-07-29 at 3 37 08 PM

### CODE REVIEW

**Review by VipulMascarenhas — COMMENTED** [review]
added minor comments, rest looks good!

File: ads/aqua/model/model.py [hunk 1]
```diff
@@ -793,29 +798,58 @@ def _validate_model_from_object_storage(
...
+                )
+
+            if not model_siblings:
+                raise AquaValueError(
+                    f"Failed to fetch the model files of {model_name} from https://huggingface.co."
+                )
+            for model_sibling in model_siblings:
+                extension = pathlib.Path(model_sibling.rfilename).suffix[1:].upper()
+                if ModelFormat.SAFETENSORS.value == extension:
+                    if AQUA_MODEL_ARTIFACT_CONFIG not in safetensors_model_files:
```
[the remark this answers is not in the document]
Comment (lu-ohai) [reply ads/aqua/model/model.py:836]: In your code, it only adds the config.json and this exactly copies the same behavior from [here](https://github.com/oracle/accelerated-data-science/blob/656e6e11e99a31e3895d6ca50c2c9c07116e2182/ads/aqua/model/model.py#L775). Could you also confirm?

File: ads/aqua/model/model.py [hunk 2]
```diff
@@ -793,29 +798,58 @@ def _validate_model_from_object_storage(
...
+
+        if download_from_hf:
+            model_siblings = []
+            try:
+                model_siblings = HfApi().model_info(
+                    repo_id=model_name
+                ).siblings
+            except Exception as e:
+                huggingface_err_message = str(e)
+                raise Exception(
```
[the remark this answers is not in the document]
Comment (lu-ohai) [reply ads/aqua/model/model.py:825]: This is the same code as the previous version

File: ads/aqua/model/model.py [hunk 3]
```diff
@@ -914,10 +955,54 @@ def _validate_model_from_object_storage(
...
+            else:
+                break
+        if i == retry:
+            raise Exception(
+                f"Could not download the model {model_name} from https://huggingface.co with message {huggingface_download_err_message}"
+            )
+        os.makedirs(local_dir, exist_ok=True)
+        # Copy the model from the cache to destination
+        snapshot_download(
+            repo_id=model_name, local_dir=local_dir, local_dir_use_symlinks=False
```
[the remark this answers is not in the document]
Comment (lu-ohai) [reply ads/aqua/model/model.py:995]: Same from the previous version. Let me remove it.

File: ads/aqua/model/model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (VipulMascarenhas) [reply ads/aqua/model/model.py:836]: you're right - I wasn't aware it was done this way. Anyway, let's go with the existing method. We can revisit later.

**Pull request merged**